client/components/saved-result.jsx

```diff
@@ -1,64 +1,211 @@
+                </div>
+                <hr className="style1 w-100" />
+              </div>
+            );
+          } else if (endpoint === 'playlists') {
+            return (
+              <div
+                key={item.playlistId}
+                data-id={item.playlistId}
+                className="row align-items-center clickable-row" >
+                <div className="col-4 ps-0">
+                  <div className="add-playlist-container d-flex">
+                    <span className="material-symbols-outlined color-icons">queue_music</span>
                   </div>
-                  <hr className="style1 w-100" />
                 </div>
-              );
-            }
-          });
-          this.setState({ result: savedList });
-        }
+                <div className="col-7">
+                  <span>{item.name}</span>
+                </div>
+                <div className="col-1">
+                  <span className="material-symbols-outlined color-icons">
+                    chevron_right
+                  </span>
+                </div>
+                <hr className="style1 w-100 mt-1" />
+              </div>
+            );
+          }
+        });
+        this.setState({ result: savedList });
       });
   }
 
-  render() {
-    const { result } = this.state;
+  handleCreatePlaylist(e) {
+    this.setState({ createPlaylist: true });
+  }
+
+  handleChange(e) {
+    const { name, value } = e.target;
+    this.setState({ [name]: value });
+  }
+
+  handleCancel() {
+    this.setState({
+      createPlaylist: false,
+      playlistName: ''
+    });
+  }
+
+  handleConfirm(e) {
+    e.preventDefault();
+    const { playlistName } = this.state;
+    const token = window.localStorage.getItem('user-jwt');
+    const options = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        'X-Access-Token': token
+      },
+      body: JSON.stringify({ playlistName })
+    };
+    fetch('/api/create/playlist', options)
+      .then(res => res.json())
+      .then(result => {
+        const savedPlaylist = result.map((item, index) => {
+          return (
+            <div key={index} className="row align-items-center clickable-row">
+              <div className="col-4 ps-0">
+                <div className="add-playlist-container d-flex">
+                  <span className="material-symbols-outlined color-icons">queue_music</span>
+                </div>
+              </div>
+              <div className="col-7">
+                <span>{item.name}</span>
+              </div>
+              <div className="col-1">
+                <span className="material-symbols-outlined color-icons">
+                  chevron_right
+                </span>
+              </div>
+              <hr className="style1 w-100 mt-1" />
+            </div>
+          );
+        });
+        this.setState({
+          result: savedPlaylist,
```

Comment:
We typically don't store React elements in state. Instead, put the data in your state and `.map()` it when rendering.